### PR TITLE
AOF now runs a routine reading from an input channel and writing to disk

### PR DESCRIPTION
**Pull request by seiflotfy**
Fixes: #131

### CODE REVIEW

File: src/storage/aof.go [hunk 1]
```diff
@@ -26,70 +27,41 @@ func NewAOF(path string) *AOF {
...
-	if err != nil {
-		return err
-	}
-	return nil
+// Run ...
+func (aof *AOF) Run() {
+	go func() {
+		for {
+			select {
+			case e := <-aof.in:
```
Comment (njpatel) [inline src/storage/aof.go]: no need for the select as there's only one option

File: src/storage/aof.go [hunk 2]
```diff
@@ -26,70 +27,41 @@ func NewAOF(path string) *AOF {
 	utils.PanicOnError(err)
 	rdr := bufio.NewReader(file)
 	wtr := bufio.NewWriter(file)
-	return &AOF{file, bufio.NewReadWriter(rdr, wtr), sync.RWMutex{}}
+	in := make(chan *Entry)
```
Comment (njpatel) [inline src/storage/aof.go]: What about making this a sized chan so there's a bit of a buffer if writing slows down? Or do you want to wait?

File: src/storage/aof.go (hunk 1 above)
Comment (seiflotfy) [inline src/storage/aof.go]: I intend to add a quit channel so it knows when to quit

File: src/storage/aof.go (hunk 2 above)
Comment (seiflotfy) [inline src/storage/aof.go]: Good call, what would a good size be though?

File: src/storage/aof.go (hunk 1 above)
Comment (njpatel) [inline src/storage/aof.go]: Makes sense

File: src/storage/aof.go (hunk 2 above)
Comment (njpatel) [inline src/storage/aof.go]: Not sure, let's start with 100 and see if we can measure it in the future under load. There's a few things we need to do to buffer writes in time, but 100 is fine for now :)

**Pull request merged**